packages/video_player/video_player/example/integration_test/video_player_test.dart

```diff
@@ -29,10 +34,53 @@ void main() {
           const Duration(seconds: 7, milliseconds: 540));
     });
 
+    testWidgets(
+      'reports buffering status',
+      (WidgetTester tester) async {
+        VideoPlayerController networkController = VideoPlayerController.network(
+          'https://flutter.github.io/assets-for-api-docs/assets/videos/bee.mp4',
+        );
+        await networkController.initialize();
+        // Mute to allow playing without DOM interaction on Web.
+        // See https://developers.google.com/web/updates/2017/09/autoplay-policy-changes
+        await networkController.setVolume(0);
+        List<VideoEventType> recordedEventTypes = [];
+        videoPlayerPlatform
+            // ignore: invalid_use_of_visible_for_testing_member
+            .videoEventsFor(networkController.textureId)
+            .listen((event) => recordedEventTypes.add(event.eventType));
+
+        await networkController.play();
+        await networkController.seekTo(const Duration(seconds: 5));
+        await tester.pumpAndSettle(_playDuration);
+        await networkController.pause();
+
+        expect(networkController.value.isPlaying, false);
+        expect(networkController.value.position,
+            (Duration position) => position > const Duration(seconds: 0));
+
+        expect(
```

Comment:
Can we use `networkController.addListener()` to test if `networkController.bufferingStart` has been set? We can even check the order of these event types too. 

This way, we don't have to expose `videoPlayerPlatform`. 

Something like:
```dart 
        final Completer<void> started = Completer();
        final Completer<void> ended = Completer();
        bool startedBuffering = false;
        networkController.addListener(() {
          if (networkController.value.isBuffering) {
            startedBuffering = true;
            started.complete();
          }
          if (startedBuffering && !networkController.value.isBuffering) {
            ended.complete();
          }
        });
```

Then we can check the state at desired places, such as
```dart
        await started;
        expect(startedBuffering, true);
```

```dart
        await ended;
        expect(startedBuffering, false);
```

(The above code is not tested)

Reply:
I did do this initially before exposing the platform but had issues, tried it again for sanity and seems to be working now 🙈 updated PR to test using this approach and remove the `videoPlayerPlatform` visibility change.